public/js/components/Scopes.js

```diff
@@ -3,30 +3,38 @@ const { bindActionCreators } = require("redux");
 const { connect } = require("react-redux");
 const ImPropTypes = require("react-immutable-proptypes");
 const actions = require("../actions");
-const { getPause, getLoadedObjects } = require("../selectors");
+const { getSelectedFrame, getLoadedObjects, getPause } = require("../selectors");
 const ObjectInspector = React.createFactory(require("./ObjectInspector"));
 const { DOM: dom, PropTypes } = React;
+const toPairs = require("lodash/toPairs");
 
 require("./Scopes.css");
 
 function info(text) {
   return dom.div({ className: "pane-info" }, text);
 }
 
+function excludeVariable(binding) {
+  if (!binding[1]) {
+    return;
+  }
+  const value = binding[1].value;
+  return value.missingArguments || value.optimizedOut;
+}
+
 // Create the tree nodes representing all the variables and arguments
 // for the bindings from a scope.
 function getBindingVariables(bindings, parentName) {
-  return bindings
-    .get("arguments").map(arg => arg.entrySeq().get(0))
-    .concat(bindings.get("variables").entrySeq())
-    .filter(binding => (!binding[1].hasIn(["value", "missingArguments"]) &&
-                        !binding[1].hasIn(["value", "optimizedOut"])))
+  const args = [].concat(...bindings.arguments.map(toPairs));
+  const variables = toPairs(bindings.variables);
+
+  return args.concat(variables)
+  .filter(binding => !excludeVariable(binding))
     .map(binding => ({
       name: binding[0],
       path: parentName + "/" + binding[0],
-      contents: binding[1].toJS()
-    }))
-    .toArray();
+      contents: binding[1]
+    }));
```

Comment:
I'd rather keep the logic of `excludeVariable` in here, it feels like over-abstracting. How come you check of `binding[1]` is truthy in the function?  You should just able tp filter out `binding[1].value.missingArguments || binding[1].value.optimizedOut;` which is pretty short.